src/app/features/dashboard/dashboard.component.ts

```diff
@@ -47,16 +49,38 @@ export class DashboardComponent {
   ];
   activeTab: MenuItem = this.tabItems[0];
 
-  constructor(private usersService: UsersService) {
+  constructor(
+    private usersService: UsersService,
+    private route: ActivatedRoute,
+    private router: Router,
+  ) {
     usersService.getCollection().then((collection) => {
       if (collection) {
         this.collection = collection;
         this.statistics = usersService.getCollectionStatistics(collection);
       }
     });
+
+    route.queryParams.pipe(takeUntilDestroyed()).subscribe((params) => {
+      if (!params) return;
+      const mediaType = params['tab'];
+      switch (mediaType) {
+        case 'movies':
+          this.activeTab = this.tabItems[1];
+          return;
+        case 'albums':
+          this.activeTab = this.tabItems[2];
+          return;
+        default:
+          this.activeTab = this.tabItems[0];
+          return;
+      }
+    });
   }
 
   changeActiveItem(event: MenuItem) {
-    this.activeTab = event;
+    this.router.navigate(['/dashboard'], {
+      queryParams: { tab: event.label?.toLowerCase() },
```

Comment:
I think the query params are good. But if you dare you could try to implement it through actual routes. Would be interesting how much more code that would require. And my gut feeling tells me that a proper routing would be a better fit here. It's like sub pages you're navigating. The query params could be replaced by providing extra routing data through `data: {activeTab: ...}` to keep the current switch.

Or better: you use an additional router outlet for this part. Also perfectly fine and would reduce the repetitive tab logic in the dashboard component (that components would go into the router)